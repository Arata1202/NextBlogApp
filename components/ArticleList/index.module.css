.main {
  /* margin-top: 20px; */
  margin-left: 0px;
}
