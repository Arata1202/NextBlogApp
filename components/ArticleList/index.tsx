import React from 'react';
import { Article } from '@/libs/microcms';
import ArticleListItem from '../ArticleListItem';
import styles from './index.module.css';
import TopSidebar from '../TopSidebar';
import Share from '../Share';
import Display from '../Adsense/display';

type Props = {
  articles?: Article[];
};

const ArticleList = ({ articles }: Props) => {
  if (!articles || articles.length === 0) {
    return (
      <div className="max-w-[85rem] sm:px-6 lg:px-8 mx-auto">
        <div className="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6">
          {/* Main Content Area */}
          <div className="lg:col-span-2">
            <div className="text-center pt-7">
              <h1 className="mt-4 text-3xl font-bold tracking-tight text-gray-900 sm:text-5xl">
                記事はまだありません
              </h1>
            </div>
          </div>
          <TopSidebar />
        </div>
      </div>
    );
  }

  return (
    <div className="max-w-[85rem] sm:px-6 lg:px-8 mx-auto">
      <div className="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6">
        {/* Main Content Area */}
        <div className="lg:col-span-2">
          <div className="FirstAd">
            <Display slot="7197259627" />
          </div>
          <ul className={`${styles.main}`}>
            <div>
              {articles.map((article) => (
                <ArticleListItem key={article.id} article={article} />
              ))}
            </div>
          </ul>
          <div className="FirstAd">
            <Display slot="1831092739" />
          </div>
          <Share />
        </div>
        <div className="mobile">
          <TopSidebar />
        </div>
      </div>
    </div>
  );
};
export default React.memo(ArticleList);
