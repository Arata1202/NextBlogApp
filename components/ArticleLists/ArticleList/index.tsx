'use client';
import React from 'react';
import { Article } from '@/libs/microcms';
import ArticleListItem from '../ArticleListItem';
import styles from './index.module.css';
import TopSidebar from '../../Sidebars/TopSidebar';
import Share from '../../Elements/Share';
import Display from '../../Adsense/Display';
import { useTheme } from 'next-themes';

type Props = {
  articles?: Article[];
};

const ArticleList = ({ articles }: Props) => {
  const { theme } = useTheme();
  if (!articles || articles.length === 0) {
    return (
      <div className="max-w-[85rem] sm:px-6 lg:px-8 mx-auto">
        <div className="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6">
          {/* Main Content Area */}
          <div className="lg:col-span-2">
            <div className="text-center pt-7">
              <h1
                className={`mt-4 text-3xl font-bold tracking-tight sm:text-5xl ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
              >
                記事はまだありません
              </h1>
            </div>
          </div>
          <TopSidebar />
        </div>
      </div>
    );
  }

  return (
    <div className="max-w-[85rem] sm:px-6 lg:px-8 mx-auto">
      <div className="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6">
        {/* Main Content Area */}
        <div className="lg:col-span-2">
          {/* <div className="FirstAd">
            <Display slot="7197259627" />
          </div> */}
          <ul className={`${styles.main}`}>
            <div>
              {articles.slice(0, 3).map((article) => (
                <ArticleListItem key={article.id} article={article} />
              ))}
            </div>
            <div className="FirstAd mb-5">
              <Display slot="9947663897" />
            </div>
            <div>
              {articles.slice(3).map((article) => (
                <ArticleListItem key={article.id} article={article} />
              ))}
            </div>
          </ul>
          <div className="FirstAd">
            <Display slot="1831092739" />
          </div>
          <Share />
        </div>
        <div className="mobile">
          <TopSidebar />
        </div>
      </div>
    </div>
  );
};
export default React.memo(ArticleList);
