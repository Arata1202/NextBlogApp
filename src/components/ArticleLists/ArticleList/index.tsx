'use client';

import { useTheme } from 'next-themes';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';
import Display from '../../Adsense/Display';
import MainContainer from '@/components/Common/Layouts/Container/MainContainer';
import ContentContainer from '@/components/Common/Layouts/Container/ContentContainer';
import ArticleCard from '@/components/Common/ArticleCard';
import TopSidebar from '../../Sidebars/TopSidebar';
import Share from '../../Elements/Share';

type Props = {
  articles: Article[];
};

export default function ArticleList({ articles }: Props) {
  const { theme } = useTheme();

  if (articles.length === 0) {
    return (
      <>
        <MainContainer>
          <ContentContainer>
            <div className="text-center py-7">
              <div
                className={`my-4 text-3xl font-bold tracking-tight sm:text-5xl ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
              >
                記事はまだありません
              </div>
            </div>
            <div className="FirstAd">
              <Display slot="1831092739" />
            </div>
            <Share />
          </ContentContainer>
          <div className="mobile">
            <TopSidebar />
          </div>
        </MainContainer>
      </>
    );
  }

  return (
    <>
      <MainContainer>
        <ContentContainer>
          <ul className={styles.main}>
            <div>
              {articles.slice(0, 3).map((article) => (
                <ArticleCard key={article.id} article={article} />
              ))}
            </div>
            <div className="FirstAd">
              <Display slot="9947663897" />
            </div>
            <div className="mt-5">
              {articles.slice(3).map((article) => (
                <ArticleCard key={article.id} article={article} />
              ))}
            </div>
          </ul>
          <div className="FirstAd">
            <Display slot="1831092739" />
          </div>
          <Share />
        </ContentContainer>
        <div className="mobile">
          <TopSidebar />
        </div>
      </MainContainer>
    </>
  );
}
