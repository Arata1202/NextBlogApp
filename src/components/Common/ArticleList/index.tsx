'use client';

import { useTheme } from 'next-themes';
import { Article } from '@/types/microcms';
import styles from './index.module.css';
import AdUnit from '../../ThirdParties/GoogleAdSense/Elements/AdUnit';
import MainContainer from '@/components/Common/Layouts/Container/MainContainer';
import ContentContainer from '@/components/Common/Layouts/Container/ContentContainer';
import ArticleCard from '../ArticleCard';
import Sidebar from '../Layouts/Sidebar';
import Share from '../Share';

type Props = {
  articles: Article[];
  recentArticles?: Article[];
};

export default function ArticleList({ articles, recentArticles }: Props) {
  const { theme } = useTheme();

  return (
    <>
      <MainContainer>
        <ContentContainer>
          {articles.length === 0 && (
            <div className="text-center py-7">
              <div
                className={`my-4 text-3xl font-bold tracking-tight sm:text-5xl ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
              >
                記事はまだありません
              </div>
            </div>
          )}
          {articles.length > 0 && (
            <ul className={styles.main}>
              {articles.slice(0, 3).map((article) => (
                <ArticleCard key={article.id} article={article} />
              ))}
              <AdUnit slot="9947663897" style={{ marginBottom: '1.25rem' }} />
              {articles.slice(3).map((article) => (
                <ArticleCard key={article.id} article={article} />
              ))}
            </ul>
          )}
          <AdUnit slot="1831092739" />
          <Share />
        </ContentContainer>
        <Sidebar recentArticles={recentArticles} />
      </MainContainer>
    </>
  );
}
