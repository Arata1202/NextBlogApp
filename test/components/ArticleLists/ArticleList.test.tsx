import { render } from '@testing-library/react';
import ArticleList from '@/components/ArticleLists/ArticleList';

describe('ArticleList', () => {
  test('スナップショット(ArticleList)', () => {
    const { asFragment } = render(<ArticleList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
