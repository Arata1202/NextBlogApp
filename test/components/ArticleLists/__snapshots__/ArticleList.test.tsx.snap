// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList スナップショット(ArticleList) 1`] = `
<DocumentFragment>
  <div
    class="max-w-[85rem] sm:px-6 lg:px-8 mx-auto"
  >
    <div
      class="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6"
    >
      <div
        class="lg:col-span-2"
      >
        <div
          class="text-center pt-7"
        >
          <h1
            class="mt-4 text-3xl font-bold tracking-tight sm:text-5xl LightTheme"
          >
            記事はまだありません
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
